### PR TITLE
Setup lint and format of Go, Proto and yaml/markdown

### PR DESCRIPTION
**Pull request by kradalby**
This PR sets up linting for all our different file types in the CI and adds a format command to make to try to improve consistency.

I have disabled a bunch of linters, some intentionally.

However there is a list of "best practice" linters that I think we should add, but we have a solid backlog to get them passing.

This PR is initial work for #219

### CODE REVIEW

File: acls.go [hunk 1]
```diff
@@ -108,7 +108,9 @@ func (h *Headscale) generateACLPolicySrcIP(u string) ([]string, error) {
 	return h.expandAlias(u)
 }
 
-func (h *Headscale) generateACLPolicyDestPorts(d string) ([]tailcfg.NetPortRange, error) {
+func (h *Headscale) generateACLPolicyDestPorts(
+	d string,
+) ([]tailcfg.NetPortRange, error) {
 	tokens := strings.Split(d, ":")
 	if len(tokens) < 2 || len(tokens) > 3 {
 		return nil, errorInvalidPortFormat
```
Comment (juanfont) [inline acls.go:115]: Is this the standard gofmt config?

Comment (juanfont) [reply acls.go:115]: I mean in general, not just this line...

File: Makefile [hunk 2]
```diff
@@ -19,7 +27,12 @@ coverprofile_html:
 	go tool cover -html=coverage.out
 
 lint:
-	golangci-lint run --fix
+	golangci-lint run --fix --timeout 10m
+
+fmt:
+	prettier --check '**/**.{ts,js,md,yaml,yml,sass,css,scss,html}'
+	golines --max-len=88 --base-formatter=gofumpt -w $(GO_SOURCES)
```
Comment (juanfont) [inline Makefile:34]: Why 88?

**Review by juanfont — DISMISSED** [review]

File: acls.go (hunk 1 above)
Comment (kradalby) [reply acls.go:115]: No, its gofumpt,which I think is standard in vscode, and then it is run through golines to break long lines.

File: Makefile (hunk 2 above)
Comment (kradalby) [reply Makefile:34]: It is based on [python black](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length) (and _[Raymond Hettinger](https://youtu.be/wf-BqAjZb8M?t=260)_), which some might hate.

The idea is to make the code more naturally fit on narrow screens, so you can have multiple files open without bad editor wrapping.

Comment (kradalby) [reply Makefile:34]: I'll adda note in the reader about code style

Comment (kradalby) [reply Makefile:34]: Done

File: acls.go (hunk 1 above)
Comment (kradalby) [reply acls.go:115]: Added note in readme.

**Review by juanfont — DISMISSED** [review]
All good!

**Comment by kradalby** [comment]
This is ready for now. This should give us a lot of helpers to keep consistent good practice.

**Comment by kradalby** [comment]
PTAL @juanfont, I think we need to have the old lint requirements removed

File: .golangci.yaml [hunk 3]
```diff
@@ -49,3 +46,15 @@ linters:
     - cyclop
     - nestif
     - wsl # might be incompatible with gofumpt
+
+linters-settings:
+  varnamelen:
```
Comment (juanfont) [inline .golangci.yaml:44]: :( this will hurt

**Review by juanfont — APPROVED** [review]
Wow. Just, wow.

File: .golangci.yaml (hunk 3 above)
Comment (kradalby) [reply .golangci.yaml:44]: I have fixed The part that hurts

**Pull request merged**